h2o-core/src/main/java/water/parser/CsvParser.java

```diff
@@ -130,26 +130,17 @@
 
         case STRING:
           if (c == quotes) {
-            if (quoteCount>1) {
-              str.addChar();
-              quoteCount--;
-            }
-
             state = COND_QUOTE;
             continue MAIN_LOOP;
           }
-          if ((!isEOL(c) || quoteCount == 1) && (c != CHAR_SEPARATOR)) {
+          if ((!isEOL(c) && c != CHAR_SEPARATOR) || quoteCount == 1) {
```

Comment:
Fixes another boung I just found. There is no reason to add characters not fitting this rule in another step. All quotes are now handled at the beginning of STRING state and will not enter this state unless quoteCount is 0 or 1. It can now never be > 1.